### DISCUSSION TITLE
How to create a moving platform that follows a user-defined path

### DISCUSSION THREAD

**Discussion by ekhunter123** [dstl/Stone-Soup#578] [Q&A] [answered]
Hi. I'm using Stone Soup on a real dataset and I have a question about the moving platforms. My data is air-to-ground and I would like to use a `MovingPlatform` class to work with the airborne `Sensor`. I know the location of the sensor at each time step. Is it possible to feed this data into the platform and make it follow the given path? I've seen the `MovementController` class but I can't figure out how to incorporate this. Any suggestions? Thanks!

**Comment by oharrald-Dstl** [comment, marked as answer]
You could use the `create_smooth_transition_models` (constant accel, constant turn) or `ConstantJerkSimulator` (constant jerk) simulators to achieve this. You pass them a sequence of timestamped states and they will attempt to generate a sequence of transition models and durations (intended to be passed to the `MultiTransitionMovingPlatform`) that will allow a platform to achieve the states in the timestamps specified.
See:
https://stonesoup.readthedocs.io/en/latest/auto_examples/Smooth_Platform_Coordinate_Transitions.html#sphx-glr-auto-examples-smooth-platform-coordinate-transitions-py

**Reply by ekhunter123** [reply]
Thank you! That looks perfect.

**Reply by ekhunter123** [reply]
That seems to be working pretty well, but I get a problem when the time difference between each platform movement increases. Here is a [gist](https://gist.github.com/ekhunter123/5912ea8141fc81a5755ac2e13fa7a57a) to show what I mean. 

I set up three platforms with the same points and same times. I set it so that it takes the platform 1 hour to get between each of the points. If I make the platform move in 100 or 500 second chunks it works fine and it follows the defined path as seen below.
![sim1](https://user-images.githubusercontent.com/59794482/151571971-5c3c4867-452a-4116-b28c-963498e2c7b5.png)

But then if I make it take 600 second steps, it suddenly veers off course about halfway through the simulation. I've tried this with a few other examples and I get the same result as soon as I increase the `sim_rate` parameter too much. Does anyone know what is causing this? 
![sim2](https://user-images.githubusercontent.com/59794482/151571972-a3684dc8-7515-4de9-87c2-e089bdbbd54f.png)

Thank you :)
